src/hotspot/share/gc/shenandoah/shenandoahFreeSet.cpp

```diff
@@ -146,14 +146,14 @@ HeapWord* ShenandoahFreeSet::try_allocate_in(ShenandoahHeapRegion* r, Shenandoah
     return NULL;
   }
 
+  try_recycle_trashed(r);
```

Comment:
I am curious why this code moved up? I do not believe there is anything wrong just curious

Reply:
This is a bug fix. We must clear the trash state and we should attempt to make this region free BEFORE asking to assign it to a generation for allocation.